src/target/riscv/riscv-013.c

```diff
@@ -4394,182 +4394,309 @@ static int write_memory_bus_v1(struct target *target, target_addr_t address,
 	return ERROR_OK;
 }
 
-static int write_memory_progbuf(struct target *target, target_addr_t address,
-		uint32_t size, uint32_t count, const uint8_t *buffer)
+static uint64_t get_from_buffer_and_log_write(target_addr_t address,
+		const uint8_t *buffer, uint32_t size)
 {
-	RISCV013_INFO(info);
+	assert(size <= 8);
+	uint64_t value = buf_get_u64(buffer, 0, 8 * size);
+	log_memory_access64(address, value, size, /*is_read*/ false);
```

Comment:
The intention is to enforce and ensure that each write is logged properly. Since the approach to writing it to the target is different in `_fill_batch()`  and `_startup()`, the logging is done on reads from buffer. If you insist, I will separate these calls, though IMHO this will make it harder to guarantee correctness of the logs such way.